### PR TITLE
Inline property documentation and SDK docs

### LINKED ISSUE #567: Better solution for tooltips
From https://github.com/maputnik/editor/pull/563#issuecomment-542620356

> The Light position and Style Renderer tooltips are not fully visible and when scrolled down, the scroll position is set back to the initial position. This issue exists also for some tooltips of the Data Sources modal.

From https://github.com/maputnik/editor/pull/563#issuecomment-542644549

> If you collapse the JSON editor on a layer editing panel it's also an issue there as well. I think we should either find a good react tooltip library or alternatively a different UI for help text. I might be better to, on hover show a ? next to the label and then on click show link to the place in the mapbox-gl docs (or our own render of the mapbox-gl docs)

tooltip_issue

### PR DESCRIPTION
**Pull request by orangemug**
I've added inline documentation instead of tooltips as outlined in #567. I've also added SDK support to the property documentation where available. 

Fixes #567 (partial fix for #427)

Demo <https://2014-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html>

Screenshot 2020-01-23 at 08 34 15

### CODE REVIEW

**Comment by pathmapper** [comment]
Oh nice, just saw this PR after leaving comments in #425 and #567.

https://github.com/maputnik/editor/issues/567#issuecomment-577598795 is still valid, there's a part of the description missing for:
https://docs.mapbox.com/mapbox-gl-js/style-spec/#layout-line-line-cap
https://docs.mapbox.com/mapbox-gl-js/style-spec/#layout-line-line-join

Any chance to add it?

**Comment by pathmapper** [comment]
These missing parts are nested in the style spec, e.g.:
https://github.com/mapbox/mapbox-gl-js/blob/master/src/style-spec/reference/v8.json#L783

**Comment by orangemug** [comment]
Good spot @pathmapper, added in latest commit.

Demo <https://2020-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html>

I've given it a quick test and it looks good, I'm a little short on time today though

**Comment by orangemug** [comment]
New build with fixes https://2031-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html

**Comment by pathmapper** [comment]
I really like the inline documentation, some findings:

- `Sources` Modal:
When adding a source, the tooltip doesn't change when the source type is changed:

image


- There a no tooltips for the `Export` modal like they are for the token fields in the `Style Settings` modal.:

image

**Comment by orangemug** [comment]
Thanks @pathmapper I've added the docs to the export modal and updated the 'type' in the sources modal 

With regards to the data source type, I've gone with a simple message instead of listing the possible values. As explained in a code comment (https://github.com/maputnik/editor/pull/607/files#diff-3648d61a30bd24d5ecf41c74cfb5fc05R170-R176) 

> Kind of a hack because the type changes, however maputnik has `1..n`
> options per type, for example 
>
>  - 'geojson' - 'GeoJSON (URL)' and 'GeoJSON (JSON)'
>  - 'raster' - 'Raster (TileJSON URL)' and 'Raster (XYZ URL)'
>
> So we just ignore the values entirely as they are self explanatory

New build <https://2035-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html>

File: src/components/modals/SettingsModal.jsx [hunk 1]
```diff
@@ -234,7 +244,7 @@ class SettingsModal extends React.Component {
         />
       </InputBlock>
 
-      <InputBlock label={"Style Renderer"} doc={"Choose the default Maputnik renderer for this style."}>
+      <InputBlock label={"Style Renderer"} fieldSpec={{doc: "Choose the default Maputnik renderer for this style."}}>
```
Comment (pathmapper) [inline src/components/modals/SettingsModal.jsx]: Maybe move this also to `fieldSpecAdditional`?

**Comment by pathmapper** [comment]
> So we just ignore the values entirely as they are self explanatory

Good solution!

Sometimes there are URLs in the docs included, do we want them to be functional or leave them like they are now?

e.g. https://docs.mapbox.com/mapbox-gl-js/style-spec/root/#sprite

grafik

**Comment by orangemug** [comment]
@pathmapper ah yeah forgot to mention that. So the docs are actually markdown, I think lets leave that for now. They are readable at the moment, I'll find a good small markdown parser (probably remark) and update in another PR.

**Comment by pathmapper** [comment]
@orangemug sure, looks good to me now. One comment pending (https://github.com/maputnik/editor/pull/607#pullrequestreview-348644678), but this doesn't need to be addressed, just thought that you wanted to move all Maputnik docs there.

**Comment by orangemug** [comment]
Thanks @pathmapper I've fixed the code review comment, merging!

**Pull request merged**